### PR TITLE
sharing 3 pr new files added

### PR DESCRIPTION
**Pull request by tsdimobile**
We updated the options according to the attached screen.

For all pages, regardless of the sharing type, there's always 'permissions' with radio buttons and 'advanced' options. See the attached sharing overview.

The radio button options from 'Permissions' are reflected on the overview page in the dropdown of each entry.

The process for personal shares is changed:

- The user enters the email of the sharing recipient. The autofill function of Nextcloud will provide suggestions. After clicking on a name, no email to the recipient is sent at this point!

- Step 1, permission dialogue is shown. After the user clicks on ‘Continue’…

- …we show the email dialogue (step2). when hitting confirm, the share is created, and an email is sent to the recipient. 

The process for link shares:

- The user clicks on ‘create link’

- Like for personal shares, a permission dialogue is shown. See the sharing overview for the appropriate permissions and advanced options

- After the user clicks on confirm, the share is created. No email is sent.

If the user activates ‘create password’, a password input field is shown right below.

If the user activates ‘Set expiration date, a date input field is shown right below.

For all folder shares, we have the options ‘File drop. With this option activated, only upload is possible.
Simulator Screen Shot - iPhone 12 Pro Max - 2021-12-30 at 17 04 35

Simulator Screen Shot - iPhone 12 Pro Max - 2021-12-30 at 17 07 49

Simulator Screen Shot - iPhone 12 Pro Max - 2021-12-30 at 17 07 57

### CODE REVIEW

**Comment by thisIsTheFoxe** [comment]
Hi @tsdimobile,
Thanks a lot for this nice PR! I really like the idea... and think it's a lot better idea than just using NCMenu for everything as I tried initially in #1799
There are still quite a few issues, but I think the first and foremost is to fix the conflicts and apply all the changes to the correct `develop` branch while being careful not to to undo any changes or bug fixes (like having activities and comments in one view).

<hr/>

Few top-level things I noticed / to consider:
- It's not possible to create more than one share link. Either:
  - Add "create new" option to the share menu
  - Always keep a static "add new share link" cell on top of all 
  - or another, better option I didn't think of..?
- This increases UX complexity / interaction cost
  - increases the number of taps by 2 (tapping on the menu to get to the detail view & tapping to apply changes)
    - And yes, number of taps is not everything and clarity is definitely more important. But I think it might be easier to just let the user tap the cell or the "..." menu and navigate directly to the detail without showing a menu before. I don't think there's a big advantage in having it. Obviously the actions would then be part of the detail view. Also also, I'm not sure if things like the link title and "Hide download" would categories as "Advanced permissions"
  - Toggle / switches should show immediate results and it would change the current behaviour (save immediately). I'm a bit unsure about this, since I can also see how an unstable internet connection can make a "apply changes" button very reasonable....

**Comment by tsdimobile** [comment]
Hi @thisIsTheFoxe 
Conflicts are resolved

**Comment by thisIsTheFoxe** [comment]
@tsdimobile Perfect, thank you.
Next, please make sure that your branch still builds and works as expected (check the `NCShare.storyboard` for your _'NCShareAdvancePermission'_ identifier). Also note, that we recently added [SwiftLint](https://github.com/realm/SwiftLint) to this project to improve readability and overall code quality. If possible, please ensure that especially new files are in compliance with our SwiftLint rules. 

Also, keep in mind my last comment about the fact that there is currently no way of creating multiple share links, is there?  I think the easiest solution would be to include it in the `NCMenu` that also includes the "Unshare" action. But feel free to   let me know if you have another idea..

**Comment by tsdimobile** [comment]
Hi @thisIsTheFoxe 
We have fixed NCShare storyboard issue and now you can create multiple share links and unshare also

**Comment by thisIsTheFoxe** [comment]
Thanks for your effort. 
As you can see, the CI fails. Please make sure:
- not to change the location or project settings concerning the `GoogleService-Info.plist`
- to fix as many linting issues as possible, especially linting errors
- to signoff every commit, for the [DCO](https://github.com/nextcloud/ios#dco-signoff)

**Comment by tsdimobile** [comment]
Hi @thisIsTheFoxe ,
We have fixed the swiftlint errors, now you can build the application.

**Review by thisIsTheFoxe — CHANGES_REQUESTED** [review]
Thanks for your work.
Please note that the CI build still fails.
Also note that there are a few minor lint issues that haven't been fixed yet.

As you probably can see, this is a rather large PR so it may take some time to get everything right. Nevertheless, thanks a lot for contributing! <3

File: iOSClient/Share/NCShare.swift [hunk 1]
```diff
@@ -275,6 +303,34 @@ class NCShare: UIViewController, UIGestureRecognizerDelegate, NCShareNetworkingD
...
+            let okAction = UIAlertAction(title: NSLocalizedString("_ok_", comment: ""), style: .default) {[weak self] (action:UIAlertAction) in
+                let password = alertController.textFields?.first?.text
+                self?.networking?.createShareLink(password: password ?? "")
+            }
+            
+            alertController.addAction(okAction)
+            
+            present(alertController, animated: true, completion:nil)
+        } else if shares.firstShareLink == nil {
+            networking?.createShareLink(password: "")
```
[the remark this answers is not in the document]
Comment (tsdimobile) [reply iOSClient/Share/NCShare.swift:327]: We did not ask for password while creating the link

**Comment by tsdimobile** [comment]
Hi @thisIsTheFoxe ,
I have did all the changes mentioned above and pushed the code.
Can you please review it and let me know if there anything required.

**Comment by thisIsTheFoxe** [comment]
Hi @tsdimobile,
Thanks a lot for your effort on this!
However, since there was no activity in Jan-Mar we decided to implement the ideas ourselves using some parts of your code as well. I suggest that in order to speed up the development of this feature #1931 should _replace_ this PR. Feel free to check it out and let us know what you think.
We'll keep this PR open for now until we reached a final decision.

**Comment by thisIsTheFoxe** [comment]
Closing this in favour of #1931

**Pull request closed without merging**